src/bp/ui-studio/src/web/views/OneFlow/sidePanel/Library/index.tsx

```diff
@@ -1,94 +1,180 @@
+        )}
+      >
+        <thead>
+          <tr>
+            <th>{title}</th>
+            <th></th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>{renderTableRows(items)}</tbody>
+      </table>
+    )
+  }
+  const renderTableRows = (items: any[]) => {
+    return (
+      <Fragment>
+        {items &&
+          items.map((item, i) => (
+            <tr key={i}>
+              <td>
+                <Button text={item.label} onClick={item.click} />
+              </td>
+              <td>
+                <Button icon="edit" onClick={item.edit} />
+              </td>
+              <td>
+                <Button icon="delete" onClick={item.delete} />
+              </td>
+            </tr>
+          ))}
+      </Fragment>
     )
   }
 
-  const onDoubleClick = (element: LibraryElement, elementType) => {
-    if (elementType === 'document') {
-      setEditItem(element.contentId)
-      setEditOpen(true)
-    }
+  const renderVariableTypes = () => {
+    return (
+      <section>
+        {renderTable(
+          'Variable Types',
+          entities
+            .filter(x => x.type !== 'system')
+            .map(x => ({
+              label: x.name,
+              click: () => {
+                setCurrentEntity(x)
+              },
+              edit: () => {
+                setCurrentNamingEntity(x)
+              },
+              delete: async () => {
+                await deleteEntity(x)
+              }
+            }))
+        )}
+        <p>
+          <Button text="Add Enumeration" onClick={() => createEntity('list')} />
+        </p>
+        <p>
+          <Button text="Add Pattern" onClick={() => createEntity('pattern')} />
+        </p>
+      </section>
+    )
   }
 
-  const filters = { text: props.filter, field: 'preview' }
-  const toggle = () => setEditOpen(false)
+  const renderModal = () => {
+    return (
+      <EntityModal
+        entity={currentEntity}
+        entities={entities}
+        updateEntity={updateEntity}
+        isOpen={currentEntity !== undefined}
+        onClose={() => setCurrentEntity(undefined)}
+      />
+    )
+  }
+  const renderNameModal = () => {
+    return (
+      <EntityNameModal
+        entity={currentNamingEntity}
+        isOpen={currentNamingEntity !== undefined}
+        onClose={() => {
+          setForceUpdate(!forceUpdate)
+          setCurrentNamingEntity(undefined)
+        }}
+      />
+    )
+  }
 
   return (
-    <div>
-      <TreeView<LibraryElement>
-        elements={props.library}
-        nodeRenderer={nodeRenderer}
-        onContextMenu={getContextMenu}
-        onDoubleClick={onDoubleClick}
-        filterText={props.filter}
-        filterProps="preview"
-      />
-      <Editor itemId={editItem} isOpen={isEditOpen} toggle={toggle}></Editor>
+    <div className={style.library}>
+      {renderVariableTypes()}
+      {renderModal()}
+      {renderNameModal()}
```

Comment:
Can you avoid "render methods" in FC components ? If you need multiple renderers, you can create new components, but we try to avoid using "render" in new components. 

There were frequents in previous "Component classes" because those spanned sometime a couple of hundred of lines.

Reply:
Ok but this menu is meant as a quick implementation just so we have some ui to test. If I spend time making 10 different components it kind of defeats the purpose of the menu.